code/ship/shiphit.cpp

```diff
@@ -694,7 +698,7 @@ float do_subobj_hit_stuff(object *ship_objp, object *other_obj, vec3d *hitpos, i
 	}
 
 	//	Shockwave damage is applied like weapon damage.  It gets consumed.
-	if ((other_obj != NULL) && (other_obj->type == OBJ_SHOCKWAVE))	// Goober5000 check for NULL
+	if (other_obj_is_shockwave)	// Goober5000 check for NULL
```

Comment:
You could remove the `// Goober5000 check for NULL` part of this line since the null check is now done earlier.